builtin/transposed_dataset.cc

```diff
@@ -54,7 +54,7 @@ struct TransposedDataset::Itl
         : dataset(dataset),
           matrix(dataset->getMatrixView()),
           index(dataset->getColumnIndex()),
-          columnCount(matrix->getColumnNames().size())
+          columnCount(dataset->getFlattenedColumnNames().size())
```

Comment:
We should add `Dataset::getFlattenedColumnCount()` here so that we don't have to create and copy the whole list, only to keep just its length.